### ISSUE TITLE
Apparent "must have the same value" false-positives when `UPDATE`ing (de)serialized Acceleration Structures

### ISSUE THREAD

**Issue by MarijnS95** [KhronosGroup/Vulkan-ValidationLayers#8590]
**Environment:**
 - OS: Linux/Windows
 - GPU and driver version: Intel Arc, amdgpu, ...
 - SDK or header version if building from repo: 1.3.290
 - Options enabled (synchronization, best practices, etc.): validation errors

**Describe the Issue**

When building an Acceleration Structure using `VK_BUILD_ACCELERATION_STRUCTURE_ALLOW_UPDATE_BIT_KHR`, then serializing it to memory using `vkCmdCopyAccelerationStructureToMemoryKHR()` and `VK_COPY_ACCELERATION_STRUCTURE_MODE_SERIALIZE_KHR`, then deserializing it using `vkCmdCopyMemoryToAccelerationStructureKHR()` and `VK_COPY_ACCELERATION_STRUCTURE_MODE_DESERIALIZE_KHR`, and finally invoking a _refit_ using `vkCmdBuildAccelerationStructuresKHR()` with `VK_BUILD_ACCELERATION_STRUCTURE_MODE_UPDATE_KHR`, the following VUIDs are raised where the validation layer does not seem to remember or have access to the original `flags`, `type` and `geometryCount` fields from `VkAccelerationStructureBuildGeometryInfoKHR` before it was serialized, and appears to assume they were all `0`:

```
Validation Error: [ VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03759 ] Object 0: handle = 0x58c967ebd190, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xa7c5450000000023, name = unbuilt blas, type = VK_OBJECT_TYPE_ACCELERATION_STRUCTURE_KHR; | MessageID = 0xa42a3268 | vkCmdBuildAccelerationStructuresKHR(): pInfos[0].mode is VK_BUILD_ACCELERATION_STRUCTURE_MODE_UPDATE_KHR, but pInfos[0].flags (VK_BUILD_ACCELERATION_STRUCTURE_ALLOW_UPDATE_BIT_KHR|VK_BUILD_ACCELERATION_STRUCTURE_PREFER_FAST_TRACE_BIT_KHR) must have the same value which was specified when srcAccelerationStructure was last built (VkBuildAccelerationStructureFlagsKHR(0)). The Vulkan spec states: For each element of pInfos, if its mode member is VK_BUILD_ACCELERATION_STRUCTURE_MODE_UPDATE_KHR, its flags member must have the same value which was specified when srcAccelerationStructure was last built (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03759)
Validation Error: [ VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03760 ] Object 0: handle = 0x58c967ebd190, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xa7c5450000000023, name = unbuilt blas, type = VK_OBJECT_TYPE_ACCELERATION_STRUCTURE_KHR; | MessageID = 0xc3b8ec1e | vkCmdBuildAccelerationStructuresKHR(): pInfos[0].mode is VK_BUILD_ACCELERATION_STRUCTURE_MODE_UPDATE_KHR, but type (VK_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL_KHR) must have the same value which was specified when srcAccelerationStructure was last built (VK_ACCELERATION_STRUCTURE_TYPE_TOP_LEVEL_KHR). The Vulkan spec states: For each element of pInfos, if its mode member is VK_BUILD_ACCELERATION_STRUCTURE_MODE_UPDATE_KHR, its type member must have the same value which was specified when srcAccelerationStructure was last built (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03760)
Validation Error: [ VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03758 ] Object 0: handle = 0x58c967ebd190, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xa7c5450000000023, name = unbuilt blas, type = VK_OBJECT_TYPE_ACCELERATION_STRUCTURE_KHR; | MessageID = 0x6f15383a | vkCmdBuildAccelerationStructuresKHR(): pInfos[0].mode is VK_BUILD_ACCELERATION_STRUCTURE_MODE_UPDATE_KHR, but geometryCount (1) must have the same value which was specified when srcAccelerationStructure was last built (0). The Vulkan spec states: For each element of pInfos, if its mode member is VK_BUILD_ACCELERATION_STRUCTURE_MODE_UPDATE_KHR, its geometryCount member must have the same value which was specified when srcAccelerationStructure was last built (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03758)
```

**Expected behavior**

No validation layer errors, unless there is documentation stating that `VK_BUILD_ACCELERATION_STRUCTURE_ALLOW_UPDATE_BIT_KHR` ASes cannot be serialized and/or cannot be refitted after deserialization.

**Valid Usage ID**
```
VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03759
VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03760
VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03758
```

**Comment by MarijnS95** [comment]
Thanks @arno-lunarg, I downloaded the layer from https://github.com/KhronosGroup/Vulkan-ValidationLayers/actions/runs/11893778937 and can confirm that this VUID no longer triggers on deserialized ASes 👍

**Comment by arno-lunarg** [comment]
Nice! Just know that the fix is essentially to give up validation. Since we don't serialize validation metadata, de-serialized acceleration structure are assumed to be valid. No easy way around that : /

**Comment by MarijnS95** [comment]
@arno-lunarg that's all fine, better have no validation than wrong validation :)

Just to note, has anything changed to timeline semaphore validation since `1.3.290`/`1.3.296`? On this download I was hitting the following validation error despite clearly seeing that a previous command buffer submit was signaling our semaphore to the expected wait value (and there's no deadlock to back this up either):

> Validation Error: [ VUID-vkQueuePresentKHR-pWaitSemaphores-03268 ] Object 0: handle = 0xb12fb2000000002c, name = Present ready 0, type = VK_OBJECT_TYPE_SEMAPHORE; Object 1: handle = 0xb9181f0000000029, type = VK_OBJECT_TYPE_SEMAPHORE; Object 2: handle = 0x125ee2133d8, type = VK_OBJECT_TYPE_QUEUE; | MessageID = 0x251f8f7a | vkQueuePresentKHR(): pPresentInfo->pWaitSemaphores[0] queue (VkQueue 0x125ee2133d8[]) is waiting on binary semaphore (VkSemaphore 0xb12fb2000000002c[Present ready 0]) that has an associated signal but it depends on timeline semaphore wait (VkSemaphore 0xb9181f0000000029[], wait value = 1) that does not have resolving signal submitted yet.
The Vulkan spec states: All elements of the pWaitSemaphores member of pPresentInfo must reference a semaphore signal operation that has been submitted for execution and any semaphore signal operations on which it depends must have also been submitted for execution (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkQueuePresentKHR-pWaitSemaphores-03268)

**Comment by arno-lunarg** [comment]
@artem-lunarg may have an answer on that

**Comment by artem-lunarg** [comment]
@MarijnS95 Yes, there was a change. We added missing validation and you probably hitting it.

This VU:
> VUID-vkQueuePresentKHR-pWaitSemaphores-03268
All elements of the pWaitSemaphores member of pPresentInfo must reference a semaphore signal operation that has been submitted for execution **and any [semaphore signal operations](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#synchronization-semaphores-signaling) on which it depends must have also been submitted for execution**

Previously we did not validated highlighted part. I know it's sounds restrictive but I clarified this with Khronos board to be sure. 

If you have a setup where swapchain binary semaphore waits on binary signal, and that binary signal depends on timeline wait, then corresponding timeline signal should be submitted before timeline wait. Although timeline wait before signal is a feature of timeline semaphores, you can't use it if it blocks binary wait. Basically binary wait can't be blocked directly or indirectly, all commands should be ready for execution without pending waits.

**Comment by MarijnS95** [comment]
@artem-lunarg yes I think I was even part of that discussion on the Khronos side because it was right around the time that I was reading the fine print on these VUs to remove an invalid signal-after-wait via `vkSignalSemaphore()`. Glad to hear that has now turned into a proper validation warning.

However, that's not what is supposed to be happening here, this is supposed to be a signal-before-wait from the same queue. If my ability to analyze code logic, and my log messages are correct, this is what should be happening:

We have a timeline semaphore T, and binary semaphores "ANI ready" and "Present ready". Two command buffers are submitted, where `cmd2` writes to our presentation image and wait for + signals the binary presentation semaphores:

```
QueueSubmit(cmd1, wait=[], signal=[T=1])
AcquireNextImageKHR(signal="ANI ready")
QueueSubmit(cmd2, wait=[T=1, "ANI ready"], signal=[T=2, "Present ready"])
QueuePresentKHR(wait="Present ready")
````

This is all happening on the same thread and on the same queue, in order.

I don't want to derail this issue much further, so if you agree that this is valid usage I'll try my best to make an API trace and/or minimal repro, and open a new issue to discuss this further (assuming I don't debunk the code by that time).

**Comment by artem-lunarg** [comment]
@MarijnS95 The code looks correct for me. I used simpler scenarios in the tests so could miss this this. Thanks, feel free to open a new issue.
